fluent/client/ws_client.go

```diff
@@ -114,12 +112,12 @@ func (wcf *DefaultWSConnectionFactory) New() (ext.Conn, error) {
 	}
 
 	conn, resp, err := dialer.Dial(wcf.URL, header)
+
 	if resp != nil && resp.Body != nil {
 		bodyBytes, readErr := io.ReadAll(resp.Body)
 		if readErr == nil {
-			bodyString := string(bodyBytes)
 			if resp.StatusCode >= 300 {
-				err = fmt.Errorf("%s. %s:%s", err, strconv.Itoa(resp.StatusCode), bodyString)
+				return nil, NewWSConnError(err, resp.StatusCode, string(bodyBytes))
```

Comment:
Thinking we should not return here, otherwise we leave the response body open. Should probably do what is already being done of assigning error and then returning at the end

Reply:
sure @danielgrist